src/Npgsql/Internal/TypeHandlers/TextHandler.cs

```diff
@@ -302,6 +310,10 @@ static unsafe void WriteCharCore(char value, NpgsqlWriteBuffer buf)
         CancellationToken cancellationToken = default)
         => buf.WriteBytesRaw(value, async, cancellationToken);
 
+    public Task Write(Enum value, NpgsqlWriteBuffer buf, NpgsqlLengthCache? lengthCache, NpgsqlParameter? parameter, bool async,
+        CancellationToken cancellationToken = default)
+        => Write(Enum.GetName(value.GetType(), value)!, buf, lengthCache, parameter, async, cancellationToken);
```

Comment:
This writes the .NET enum name (PascalCase) as a string; I'm not sure how really useful that is - though this is a very opt-in feature (NpgsqlDbType.String must be specified) so maybe that's OK.

Note that we also have the global name translator (as configured via NpgsqlDataSource.DefaultNameTranslator), which we can consider applying here. That would be consistent with the way we handle mapped enums, though on the other hand mapped enums are mapped to a PG **type** (where the convention is snake_case), whereas here's we're mapping the name to a string value, where maybe that convention makes less sense...

Not sure what I think about it all...